### PR TITLE
feat: support specify the position when add column

### LINKED ISSUE #10685: Feature: Support change the order of column when add column
**Summary**
is means like : alter table xxx add column xxx **after** xxx
for example:
sync data from mysql to databend,if mysql add column after the specific column, i want add the column to databend automatically by writing some simple code

### PR DESCRIPTION
**Pull request by zhyass**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary
Syntax:
```sql
ALTER TABLE tbl_name ADD COLUMN col_name column_definition [FIRST | AFTER col_name]
```
The default is to add the column last. 
```
mysql> create table t(c int);
Query OK, 0 rows affected (0.04 sec)

mysql> insert into t values(1);
Query OK, 1 row affected (0.11 sec)

mysql> alter table t add column a float default 0.1 comment 'first' first;
Query OK, 0 rows affected (0.04 sec)

mysql> insert into t values(0.2, 2);
Query OK, 1 row affected (0.04 sec)

mysql> alter table t add column b string default 'b' comment 'after' after a;
Query OK, 0 rows affected (0.13 sec)

mysql> insert into t values(0.3, 'c', 3);
Query OK, 1 row affected (0.11 sec)

mysql> select * from t order by a;
+------+------+------+
| a    | b    | c    |
+------+------+------+
|  0.1 | b    |    1 |
|  0.2 | b    |    2 |
|  0.3 | c    |    3 |
+------+------+------+
3 rows in set (0.07 sec)
Read 3 rows, 229.00 B in 0.023 sec., 130.91 rows/sec., 9.76 KiB/sec.

mysql> show create table t;
+-------+--------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                   |
+-------+--------------------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `a` FLOAT DEFAULT 0.1 COMMENT 'first',
  `b` VARCHAR DEFAULT 'b' COMMENT 'after',
  `c` INT
) ENGINE=FUSE |
+-------+--------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.05 sec)
Read 0 rows, 0.00 B in 0.034 sec., 0 rows/sec., 0.00 B/sec.
```

- Closes #10685

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/12352)

### CODE REVIEW

File: src/meta/app/src/schema/table.rs [hunk 1]
```diff
@@ -250,9 +251,23 @@ impl TableMeta {
...
+        index: FieldIndex,
+    ) -> Result<()> {
+        let mut new_schema = self.schema.as_ref().to_owned();
+        new_schema.add_column(field, index)?;
+        self.schema = Arc::new(new_schema);
+        self.field_comments.insert(index, comment.to_owned());
+        Ok(())
+    }
+
     pub fn drop_column(&mut self, column: &str) -> Result<()> {
```
Comment (lichuang) [inline src/meta/app/src/schema/table.rs:267]: why change `drop_column` return type, it seems that there is no place to use this API this pr.

Comment (zhyass) [reply src/meta/app/src/schema/table.rs:267]: > drop_column

Because we need use the fieldIndex to remove the comment in `field_comments`

Comment (b41sh) [reply src/meta/app/src/schema/table.rs:267]: index is used to remove `field_comments`, which is not handled before.

**Review by b41sh — APPROVED** [review]

**Review by lichuang — APPROVED** [review]

**Comment by BohuTANG** [comment]
cc @soyeric128 for document, thanks.

**Comment by soyeric128** [comment]
> cc @soyeric128 for document, thanks.

Documented: https://databend.rs/doc/sql-commands/ddl/table/alter-table-column

**Pull request merged**